storage/local/storage.go

```diff
@@ -212,10 +212,18 @@ func (s *memorySeriesStorage) Start() error {
 		return err
 	}
 	s.persistence = p
+	go s.persistence.run()
+
+	pclose := func() {
```

Comment:
But that would either happen in a narrower scope or it would create a compile error...

The actual problem is if somebody returns an error directly, without using err.

For that, you could make the returned error a named return. That's a common idiom for deferred function that want to inspect return values.